third_party/amd/include/Dialect/TritonAMDGPU/IR/TritonAMDGPUOps.td

```diff
@@ -152,6 +152,22 @@ def InstructionSchedHint : TT_AMDGPU_Op<"instruction_sched_hint", []> {
   let assemblyFormat = [{ attr-dict }];
 }
 
+def condBarrierOp :
+  TT_AMDGPU_Op<"cond_barrier">,
+  Arguments<(ins I1:$pred)> {
+  let summary = "set barrier conditionally";
+
+  let description = [{
+      condBarrierOp sets barrier instruction only when the given argument is true.
+      This provides a way to synchronize partial threads in a block, deliberately
+      diverges the execution sequences. However, user should guarantee all threads
+      converge at the end by calling condBarrierOp(true) with the remaining threads.
+      NB. This doesn't set memory fence.
```

Comment:
Could you also add some lines like "Conceptually this is similar to having a barrier inside an if statement. This op allows us to avoid blocking the whole block when suitable to help scheduling."

Reply:
Sure, that makes sense to clarify the usage, will address it in the new PR